internal/sshtunneler/tunneler.go

```diff
@@ -49,6 +48,11 @@ type SSHDial interface {
 	Dial(conn net.Conn, username string, privateKey gossh.Signer, hostKeyCallback gossh.HostKeyCallback) (*gossh.Client, error)
 }
 
+// Clock defines an interface for getting the current time.
+type Clock interface {
```

Comment:
The reason I created this interface is because when we generate mocks for this package, we don't have to create 2 files, one with mocks for the interfaces defined in this `sshtunneler` package and another file for the Clock interface declared in `github.com/juju/juju`. And the change to longer bring in the `github.com/juju/clock` package just for the clock's interface definition definitely seems a nice plus.